ee/tools/puppet/fleetdm/lib/puppet/functions/fleetdm/preassign_profile.rb

```diff
@@ -14,7 +14,8 @@ def preassign_profile(profile_identifier, host_uuid, template, group = 'default'
     host = call_function('lookup', 'fleetdm::host')
     token = call_function('lookup', 'fleetdm::token')
     client = Puppet::Util::FleetClient.new(host, token)
-    response = client.preassign_profile(host_uuid, template, group)
+    run_identifier = closure_scope.catalog.catalog_uuid || Puppet[:node_name_value]
```

Comment:
Just curious, where is `closure_scope` defined?

Reply:
It's kind of magic, docs here: https://www.puppet.com/docs/puppet/7/functions_ruby_implementation.html#functions_ruby_implementation-accessing-variables